apps/files_sharing/lib/Controller/ShareAPIController.php

```diff
@@ -967,9 +969,9 @@ private function getLinkSharePermissions(?int $permissions, ?bool $legacyPublicU
 				: Constants::PERMISSION_READ;
 		}
 
-		// TODO: It might make sense to have a dedicated setting to allow/deny converting link shares into federated ones
 		if ($this->hasPermission($permissions, Constants::PERMISSION_READ)
-			&& $this->shareManager->outgoingServer2ServerSharesAllowed()) {
+			&& $this->shareManager->outgoingServer2ServerSharesAllowed()
+			&& $this->appConfig->getValueBool('core', 'shareapi_allow_federation_on_public_shares', true)) {
```

Comment:
Can we?

```suggestion
			&& $this->appConfig->getValueBool('core', ConfigLexicon::SHAREAPI_ALLOW_FEDERATION_ON_PUBLIC_SHARES, true)) {
```

Reply:
Also, shouldn't it be bonded to `files_sharing`?
```suggestion
			&& $this->appConfig->getValueBool('files_sharing', 'shareapi_allow_federation_on_public_shares', true)) {
```